### DISCUSSION TITLE
How does collectors.php have access to the service() function?

### DISCUSSION THREAD

**Discussion by alexisLefebvre** [symfony/symfony#47055] [Q&A] [answered]
I wonder how the function `service()` can be called here:

https://github.com/symfony/symfony/blob/ea68966e914a257c2df551160e02e658b44864cd/src/Symfony/Bundle/FrameworkBundle/Resources/config/collectors.php#L33

It's defined here:

https://github.com/symfony/symfony/blob/ea68966e914a257c2df551160e02e658b44864cd/src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php#L109

It looks like the `service()` method is available globally, but I don't understand how. Is it declared through Composer? Can someone please explain?

**Comment by HypeMC** [comment, marked as answer]
The `service()` function is not globally available, it becomes available when the `ContainerConfigurator` [class is autoloaded](https://github.com/symfony/symfony/blob/c02b7049d7230cb49dd41b9d66d4e793c14ccac0/src/Symfony/Component/DependencyInjection/Loader/PhpFileLoader.php#L149).

**Reply by alexisLefebvre** [reply]
Thanks for the explanation.
